ts/features/idpay/onboarding/screens/InitiativeDetailsScreen.tsx

```diff
@@ -97,89 +99,106 @@ const BeforeContinueBody = (props: BeforeContinueBodyProps) => {
 
 const InitiativeDetailsScreen = () => {
   const route = useRoute<InitiativeDetailsRouteProps>();
-
-  const { serviceId } = route.params;
-
-  const service = pipe(
-    pot.toOption(
-      useIOSelector(serviceByIdSelector(serviceId as ServiceId)) || pot.none
-    ),
-    O.toUndefined
-  );
-
   const onboardingMachineService = useOnboardingMachineService();
   const [state, send] = useActor(onboardingMachineService);
+  const {
+    scrollViewRef,
+    onScrollViewSizeChange,
+    scrollToEnd,
+    handleIsScrollEnd,
+    requiresScrolling,
+    setMarkdownIsLoaded
+  } = useInitiativeDetailsScrolling();
 
   const isLoading = state.tags.has(LOADING_TAG);
   const isAcceptingTos = state.tags.has(UPSERTING_TAG);
-
-  const handleGoBackPress = () => {
-    send({ type: "QUIT_ONBOARDING" });
-  };
-
-  const handleContinuePress = () => {
-    send({ type: "ACCEPT_TOS" });
-  };
+  const { serviceId } = route.params;
 
   React.useEffect(() => {
     send({
       type: "SELECT_INITIATIVE",
       serviceId
     });
   }, [send, serviceId]);
+  const handleGoBackPress = () => {
+    send({ type: "QUIT_ONBOARDING" });
+  };
+  const handleContinuePress = () =>
+    requiresScrolling ? scrollToEnd() : send({ type: "ACCEPT_TOS" });
 
   // TODO show initaitveID for testing purposes
-  const content = pipe(
+  const headerContent = pipe(
     O.fromNullable(state.context.initiative),
     O.map(initiative => `Initiative ID: ${initiative.initiativeId}`),
     O.toUndefined
   );
+  const service = pipe(
+    pot.toOption(
+      useIOSelector(serviceByIdSelector(serviceId as ServiceId)) || pot.none
+    ),
+    O.toUndefined
+  );
+
+  const screenContent = () => (
+    <SafeAreaView style={IOStyles.flex}>
+      <ScrollView
+        onContentSizeChange={onScrollViewSizeChange}
+        onScroll={({ nativeEvent }) => handleIsScrollEnd(nativeEvent)}
+        scrollEventThrottle={400}
+        ref={scrollViewRef}
+        style={IOStyles.flex}
+      >
+        <View style={IOStyles.horizontalContentPadding}>
+          {service !== undefined && (
+            <InitiativeOrganizationHeader {...toUIService(service)} />
+          )}
+          {headerContent && <Text>{headerContent}</Text>}
+          <VSpacer size={16} />
+          {state.context.initiative?.description !== undefined ? (
+            <Markdown onLoadEnd={setMarkdownIsLoaded}>
+              {state.context.initiative.description}
+            </Markdown>
+          ) : (
+            setMarkdownIsLoaded()
+          )}
```

Comment:
Isn't there a better way to do this? 
Maybe we could add a parameter to the `useInitiativeDetailsScrolling` hook which can be used to determine the `markdownIsLoadingRef` initial value.